compyle/utils.py

```diff
@@ -24,3 +25,30 @@ def getsource(obj):
             return obj.source
         else:
             raise
+
+
+class ArgumentParser(argparse.ArgumentParser):
+    '''Standard argument parser for compyle applications.
+    Includes arguments for backend, openmp and use_double
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # setup standard arguments
+        self.add_argument(
+            '-b', '--backend', action='store', dest='backend', default='cython',
+            help='Choose the backend.'
+        )
+        self.add_argument(
+            '--openmp', action='store_true', dest='openmp', default=False,
+            help='Use OpenMP.'
+        )
+        self.add_argument(
+            '--use-double', action='store_true', dest='use_double',
+            default=False, help='Use double precision on the GPU.'
+        )
+        self.add_argument(
+            '--suppress-warnings', action='store_true',
+            dest='suppress_warnings',
+            default=False, help='Suppress warnings'
+        )
```

Comment:
Perhaps a process options should be made available, or perhaps `set_config_options()` which sets up the configuration based on these options?